internal/restorer/progressprinter/progressprinter_test.go

```diff
@@ -0,0 +1,251 @@
+	printer.LogProgress("test", firstBytesWritten, fileSize)
+
+	printer.LogProgress("test", secondBytesWritten, fileSize)
+
+	actualBytesWritten := printer.progressInfoMap["test"].bytesWritten
+	actualAllBytesWritten := printer.allBytesWritten
+	if actualBytesWritten != expectedBytesWritten {
+		t.Errorf("Error on Subsequent Progress. Wrong bytesWritten value. Have %v, wanted %v", actualBytesWritten, expectedBytesWritten)
+	}
+	if actualAllBytesWritten != expectedAllBytesWritten {
+		t.Errorf("Error on Subsequent Progress. Wrong allBytesWritten value. Have %v, wanted %v", actualAllBytesWritten, expectedBytesWritten)
+	}
+}
+
+func TestLastProgressOnAFile(t *testing.T) {
+	fileSize := int64(100)
+	printer := New()
+	printer.AddFile(fileSize)
+	printer.LogProgress("test", int64(30), fileSize)
+	printer.LogProgress("test", int64(35), fileSize)
+
+	printer.LogProgress("test", int64(35), fileSize)
+
+	actualSize := len(printer.progressInfoMap)
+	actualFinished := printer.filesFinished
+	actualAllBytesWritten := printer.allBytesWritten
+	if actualSize != 0 {
+		t.Errorf("Error on Last Progress. Wrong map size. Have %v, wanted 0", actualSize)
+	}
+	if actualFinished != 1 {
+		t.Errorf("Error on Last Progress. Wrong filesFinished. Have %v, wanted 1", actualFinished)
+	}
+	if actualAllBytesWritten != fileSize {
+		t.Errorf("Error on Last Progress. Wrong AllBytesWritten. Have %v, wanted %v", actualAllBytesWritten, fileSize)
+	}
+}
+
+func TestLastProgressOnLastFile(t *testing.T) {
+	fileSize := int64(100)
+	printer := New()
+	printer.AddFile(fileSize)
+	printer.AddFile(50)
+	printer.LogProgress("test1", int64(50), int64(50))
+	printer.LogProgress("test2", int64(50), fileSize)
+	expectedAllBytesWritten := printer.allBytesTotal
+	expectedFinished := printer.filesTotal
+
+	printer.LogProgress("test2", int64(50), fileSize)
+
+	actualFinished := printer.filesFinished
+	actualAllBytesWritten := printer.allBytesWritten
+	if actualFinished != expectedFinished {
+		t.Errorf("Error on Last Progress of all. Wrong filesFinished. Have %v, wanted %v", actualFinished, expectedFinished)
+	}
+	if actualAllBytesWritten != expectedAllBytesWritten {
+		t.Errorf("Error on Last Progress of all. Wrong AllBytesWritten. Have %v, wanted %v", actualAllBytesWritten, expectedAllBytesWritten)
+	}
+}
+
+func TestPrintingWithSpecificValues(t *testing.T) {
+	printer := New()
+	printer.filesTotal = 10
+	printer.filesFinished = 6
+	printer.allBytesTotal = 1024
+	printer.allBytesWritten = 512
+	expectedPrint := "\r\033[K[0:01]  6 / 10 Files,  512 B / 1.00 KiB,  50.00 %  "
+	var writer bytes.Buffer
+	progresswriter = &writer
+	time.Sleep(time.Second)
+
+	print(printer)
+
+	printed := writer.String()
+	if printed != expectedPrint {
+		t.Errorf("Printer Error.\nHave   %v\nwanted %v",
+			strings.Replace(strings.Replace(printed, "\033", "\\033", -1), "\r", "\\r", -1), 
+			strings.Replace(strings.Replace(expectedPrint, "\033", "\\033", -1), "\r", "\\r", -1))
```

Comment:
%q does escaping for you:

```suggestion
		t.Errorf("Printer Error.\nHave   %q\nwanted %q", printed, expectedPrint)
```